OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListFragment.java

```diff
@@ -126,12 +134,28 @@ public void onLoadFinished(Loader<AsyncTaskResult<List<EnrolledCoursesResponse>>
             if (result.getEx() instanceof AuthException) {
                 loginPrefs.clear();
                 getActivity().finish();
-            } else if (result.getEx() instanceof HttpStatusException &&
-                    ((HttpStatusException) result.getEx()).getStatusCode() == HttpStatus.UNAUTHORIZED) {
-                environment.getRouter().forceLogout(
-                        getContext(),
-                        environment.getAnalyticsRegistry(),
-                        environment.getNotificationDelegate());
+            } else if (result.getEx() instanceof HttpStatusException) {
+                final HttpStatusException httpStatusException = (HttpStatusException) result.getEx();
+                switch (httpStatusException.getStatusCode()) {
+                    case HttpStatus.UNAUTHORIZED:{
+                        environment.getRouter().forceLogout(getContext(),
+                                environment.getAnalyticsRegistry(),
+                                environment.getNotificationDelegate());
+                        break;
+                    }
+                    default: {
+                        errorNotification.showError(R.string.network_connected_error,
```

Comment:
Are not we unconditionally expecting exception cause due to internet connectivity here. Suppose to take error message from `ErrorUtils.getErrorMessage` function.